In [ ]:
import numpy as np

In [ ]:
number_of_visitors = [ 150, 200, 180, 220, 190, 250, 170, 1600, 210, 230]

In [ ]:
number_of_visitors.sort()
number_of_visitors

[150, 170, 180, 190, 200, 210, 220, 230, 250, 1600]

In [ ]:
range_number_of_visitors = max(number_of_visitors) - min(number_of_visitors)
print("діапазон (range) кількості відвідувачів:", range_number_of_visitors, "одиниць")

діапазон (range) кількості відвідувачів: 1450 одиниць


Знайди квартилі (Q1, Q2, Q3) для кількості відвідувачів. Пам'ятай, що потрібно попередньо відсортувати вибірку.

Розрахуй міжквартильний діапазон (IQR).

In [ ]:
number_of_visitors_left = number_of_visitors[: len(number_of_visitors) // 2]
number_of_visitors_right = number_of_visitors[len(number_of_visitors) // 2 :]

q1 = np.median(number_of_visitors_left)
q2 = np.median(number_of_visitors)
q3 = np.median(number_of_visitors_right)

print(f"Q1: {q1}")
print(f"Q2: {q2}")
print(f"Q3: {q3}")

iqr = q3 - q1
print(f"\nміжквартильний діапазон: {iqr}")

Q1: 180.0
Q2: 205.0
Q3: 230.0

міжквартильний діапазон: 50.0


Обчисли стандартне відхилення для кількості відвідувачів.

In [ ]:
std_dev = np.std(number_of_visitors)
print("стандартне відхилення", round(std_dev))

стандартне відхилення 421


На основі отриманих результатів проаналізуй:
наскільки стабільними були відвідування вебсайту протягом цього періоду;
які типові значення кількості відвідувачів;
чи були екстремальні значення.

Діапазон - 1450 та стандартне відхилення - 421 свідчать про значні коливання кількості відвідувачів.

Дані сильно розкидані, це підтверджується високим показником стандартного відхилення порівняно з квартильними значеннями.

Типові значення відвідування - медіана - 205 відвідувачів, та інтервал між q1 та q3 - 180-230 відвідувачів.

Екстремальні значення відвівдування до 180 та більше 250